pkg/dataplane/identity.go

```diff
@@ -69,14 +82,122 @@ func (c CredentialsObject) GetCredential(requestedResourceID string) (*azidentit
 
 // Get an AzIdentity credential for the given nested credential object
 // Clients can use the credential to get a token for the user-assigned identity
-func (n NestedCredentialsObject) GetCredential() (*azidentity.ClientCertificateCredential, error) {
+func (n *NestedCredentialsObject) GetCredential() (*azidentity.ClientCertificateCredential, error) {
 	if n.Values.ResourceID != nil {
 		return getClientCertificateCredential(n.Values, n.Cloud)
 	}
 
 	return nil, errResourceIDNotFound
 }
 
+func (n *NestedCredentialsObject) ReloadCredntialsOnChange() error {
+	if err := n.File.checkFileExists(); err != nil {
+		return err
+	}
+
+	err := n.initializeWatcher()
```

Comment:
Tightly controlling the method constructor as I suggest in my other comment means there is no need to be this careful about initialization, checking for already-initialized values, `sync.Once`, etc.